app/main.py

```diff
@@ -1,8 +1,28 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self  # Add the person to the people dictionary
+
+    def __repr__(self) -> str:
+        return f"Person(name={self.name}, age={self.age})"
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_list = []
+
+    for person_data in people:
+        person_instance = Person(name=person_data["name"],
+                                 age=person_data["age"])
+        person_list.append(person_instance)
+
+    for person_data in people:
+        person_instance = Person.people[person_data["name"]]
+        if "wife" in person_data and person_data["wife"]:
+            person_instance.wife = Person.people[person_data["wife"]]
+        if "husband" in person_data and person_data["husband"]:
```

Comment:
you can just use __get()__ instead of that __and__ condition, and also you can use walrus operetor := in __if__ condition, so you don't need to call person_data["wife"] twice, same for husband

Reply:
good comment, agree!